src/main/java/org/phyloref/jphyloref/commands/WebserverCommand.java

```diff
@@ -0,0 +1,453 @@
+import org.semanticweb.owlapi.model.OWLNamedIndividual;
+import org.semanticweb.owlapi.model.OWLOntology;
+import org.semanticweb.owlapi.model.OWLOntologyCreationException;
+import org.semanticweb.owlapi.model.OWLOntologyLoaderConfiguration;
+import org.semanticweb.owlapi.model.OWLOntologyManager;
+import org.semanticweb.owlapi.reasoner.BufferingMode;
+import org.semanticweb.owlapi.rio.RioOWLRDFConsumerAdapter;
+import org.semanticweb.owlapi.search.EntitySearcher;
+import org.semanticweb.owlapi.util.AnonymousNodeChecker;
+import org.semanticweb.owlapi.util.VersionInfo;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.NanoHTTPD.Response.Status;
+import fi.iki.elonen.NanoHTTPD.Response.IStatus;
+import uk.ac.manchester.cs.jfact.JFactReasoner;
+import uk.ac.manchester.cs.jfact.kernel.options.JFactReasonerConfiguration;
+
+/**
+ * Sets up a webserver that allows reasoning over phyloreferences
+ * over HTTP.
+ *
+ * At the moment, we implement a simple API, in which:
+ *  /version:   returns a JSON object with information on the version of
+ *              JPhyloRef and the reasoner being used.
+ *  /reason:    expects a form upload with a 'jsonld' element containing a file
+ *              upload of a JSON-LD file representing an ontology to test.
+ *              This command will reason over the ontology and return a JSON
+ *              dictionary, in which the keys are the IRIs for each phyloreference,
+ *              and the values are lists of the IRIs of each node matched by that
+ *              phyloreference.
+ *
+ * @author Gaurav Vaidya <gaurav@ggvaidya.com>
+ *
+ */
+public class WebserverCommand implements Command {
+  /**
+   * This command is named "webserver". It should be
+   * invoked as "java -jar jphyloref.jar webserver ..."
+   */
+  @Override
+  public String getName() {
+    return "webserver";
+  }
+
+  /**
+   * A description of the Webserver command.
+   *
+   * @return A description of this command.
+   */
+  @Override
+  public String getDescription() {
+    return "Set up a webserver to allow reasoning of phyloreferences over HTTP.";
+  }
+
+  /**
+   * Add command-line options specific to this command.
+   *
+   * @param opts The command-line options to modify for this command.
+   */
+  @Override
+  public void addCommandLineOptions(Options opts) {
+    opts.addOption(
+      "h", "host", true,
+      "The hostname to listen to HTTP connections on (default: 'localhost')"
+    );
+    opts.addOption(
+      "p", "port", true,
+      "The TCP port to listen to HTTP connections on (default: 8080)"
+    );
+  }
+
+  /**
+   * Set up a webserver to listen on the provided hostname and port (or their defaults).
+   *
+   * @param cmdLine The command line options provided to this command.
+   */
+  @Override
+  public void execute(CommandLine cmdLine) throws RuntimeException {
+      String hostname = cmdLine.getOptionValue("host", "localhost");
+      String portString = cmdLine.getOptionValue("port", "8080");
```

Comment:
Shouldn't this be on a more unique port?

Reply:
Yes, it should! I used random.org to pick TCP port 34214, which [IANA confirms is unassigned](http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt).